lib/internal/crypto/random.js

```diff
@@ -119,6 +120,51 @@ function randomFill(buf, offset, size, cb) {
   _randomBytes(buf, offset, size, wrap);
 }
 
+// Largest integer we can read from a buffer.
+// e.g.: Buffer.from("ff".repeat(6), "hex").readUIntBE(0, 6);
+const RAND_MAX = 281474976710655;
+
+function randomInt(min, max, cb) {
+  if (typeof cb !== 'function') {
+    throw new ERR_INVALID_CALLBACK(cb);
+  }
+  if (!Number.isInteger(min)) {
+    throw new ERR_INVALID_ARG_TYPE('min', 'integer', min);
+  }
+  if (!Number.isInteger(max)) {
+    throw new ERR_INVALID_ARG_TYPE('max', 'integer', min);
+  }
+  if (!(max >= min)) {
+    throw new ERR_OUT_OF_RANGE('max', `>= ${min}`, max);
+  }
+  if (!(max <= RAND_MAX)) {
+    throw new ERR_OUT_OF_RANGE('max', `<= ${RAND_MAX}`, max);
+  }
+
+  // First we generate a random int between [0..bound)
+  const bound = max - min;
+  const range = bound - 1;
+  const excess = RAND_MAX % range;
+  const randLimit = RAND_MAX - excess;
+
+  const pickAttempt = () => {
+    randomBytes(6, (err, bytes) => {
```

Comment:
Why a hardcoded value of `6`? If the end/max value is small, this will end up having to do a lot of extra work if it continues to get a bunch of numbers outside of the desired range.

Reply:
@mscdex Because it’s easier to get

> Why a hardcoded value of `6`? 

I would keep it if performance doesn’t significantly differ between generating 6 bytes of random data and less than 6 bytes – and I would be very surprised if it did.

> If the end/max value is small, this will end up having to do a lot of extra work if it continues to get a bunch of numbers outside of the desired range.

If we generate less than 6 bytes, i.e. start with a smaller integer here, then the chance of having to re-pick will be higher. This is saving extra work, not generating it.

Reply:
> If we generate less than 6 bytes, i.e. start with a smaller integer here, then the chance of having to re-pick will be higher. This is saving extra work, not generating it.

Yes, this is the reason I hardcoded it to 6 bytes. I guess there's a lot of room for optimization but wanted to keep the first version simple.